### ISSUE TITLE
What is the supported sequence length of bert-tiny and bert-small?

### ISSUE THREAD

**Issue by John-8704** [tensorflow/hub#742]
Here are few bert models I came across recently but couldn't figure out the maximum supported sequence length, Is it 512 tokens per sentence?

Does these models have the same 512 token limit as the original bert-base model

```
L=2 | 2/128 (BERT-Tiny) | 2/256 | 2/512 | 2/768
L=4 | 4/128 | 4/256 (BERT-Mini) | 4/512 (BERT-Small) | 4/768

```

**Comment by John-8704** [comment]
@arghyaganguly , Let's say I found a model called `Bert_en_uncased_L-2_H-512_A-8` in tensorflow hub,

Here's the example code that they have given [here](Bert_en_uncased_L-2_H-512_A-8) ,

```
text_input = tf.keras.layers.Input(shape=(), dtype=tf.string)
preprocessor = hub.KerasLayer(
    "https://tfhub.dev/tensorflow/bert_en_uncased_preprocess/3")
encoder_inputs = preprocessor(text_input)
encoder = hub.KerasLayer(
    "https://tfhub.dev/tensorflow/small_bert/bert_en_uncased_L-2_H-512_A-8/1",
    trainable=True)
outputs = encoder(encoder_inputs)
pooled_output = outputs["pooled_output"]      # [batch_size, 512].
sequence_output = outputs["sequence_output"]  # [batch_size, seq_length, 512].
```

Here, the preprocessor layer is padding my sequence length to 128 tokens, can I know why this is happening if any bert model can support sequence length of 512.

How can I overcome this, which tokenizer should I use to support 512, any pointers on this will be really helpful.

**Comment by akhorlin** [comment]
You can customize the sequence length for the preprocessing layers. Here is an [example](https://www.tensorflow.org/tutorials/text/solve_glue_tasks_using_bert_on_tpu#preprocess_the_text) (check out make_bert_preprocess_model() implementation).
